### PR TITLE
DRILL-5971: Fix INT64, INT32 logical types in complex parquet reader

### PR DESCRIPTION
**Pull request by parthchandra**
Added support for the INT64 and INT32 logical types in the complex parquet reader

### CODE REVIEW

**Comment by arina-ielchiieva** [comment]
@vvysotskyi please review.

**Comment by vvysotskyi** [comment]
LGTM, +1. Except `INT_32` and `INT_64` parquet can contain other logical numeric types: `UINT_8`, `UINT_16`, `UINT_32`, `UINT_64`, `INT_8`, `INT_16`[1]. It would be great to add support for these logical types in this pull request or create another Jira for this.

[1] https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#numeric-types

**Comment by parthchandra** [comment]
Updated the PR to include all the remaining logical int types. While generating the test file for the unit test, found that some of the types were missing from the fast parquet reader so added those as well.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ColumnReaderFactory.java [hunk 1]
```diff
@@ -138,6 +138,8 @@
                 return new ParquetFixedWidthDictionaryReaders.DictionaryBigIntReader(recordReader, allocateSize, descriptor, columnChunkMetaData, fixedLength, (BigIntVector) v, schemaElement);
```
[the remark this answers is not in the document]
Comment (parthchandra) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ColumnReaderFactory.java]: OK I'll try to add these. BTW, I realized that the test files that I added for the unit tests are not annotated, so I'll need to fix those as well!

**Comment by parthchandra** [comment]
@vvysotskyi please review - 
added fixes for date time and interval types, fixed the reading of dictionary encoded fixed width array types (including int96),  fixed object representation of interval types (used by unit tests and sqlline/jdbc), and  added a new test file generator that generates data for all (primitive) logical types in either flat or hierarchical schemas. Added those files as unit tests for the logical types.

File: exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetComplex.java [hunk 2]
```diff
@@ -193,4 +200,168 @@ public void notxistsField() throws Exception {
...
+        "_DATE_int32" ,
+        "_TIME_MILLIS_int32" ,
+        "_TIMESTAMP_MILLIS_int64" ,
+        "_INTERVAL_fixed_len_byte_array_12" ,
+        "_INT96_RAW"
+
+    };
+    testBuilder()
+        .sqlQuery(query)
+        .ordered()
```
Comment (vvysotskyi) [inline exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetComplex.java]: unOrdered

**Comment by vvysotskyi** [comment]
@parthchandra, thanks for the pull request! LGTM, +1.

**Pull request closed without merging**